prow/scripts/provision-vm-compass-ord.sh

```diff
@@ -0,0 +1,127 @@
+    log::info "Removing instance compass-ord-test-${RANDOM_ID}"
+    gcloud compute instances delete --zone="${ZONE}" "compass-ord-test-${RANDOM_ID}" || true ### Workaround: not failing the job regardless of the vm deletion result
+    exit $ARG
+}
+
+function testCustomImage() {
+    CUSTOM_IMAGE="$1"
+    IMAGE_EXISTS=$(gcloud compute images list --filter "name:${CUSTOM_IMAGE}" | tail -n +2 | awk '{print $1}')
+    if [[ -z "$IMAGE_EXISTS" ]]; then
+        log::error "${CUSTOM_IMAGE} is invalid, it is not available in GCP images list, the script will terminate ..." && exit 1
+    fi
+}
+
+gcp::authenticate \
+    -c "${GOOGLE_APPLICATION_CREDENTIALS}"
+
+RANDOM_ID=$(openssl rand -hex 4)
+
+LABELS=""
+if [[ -z "${PULL_NUMBER}" ]]; then
+    LABELS=(--labels "branch=$PULL_BASE_REF,job-name=compass-ord")
+else
+    LABELS=(--labels "pull-number=$PULL_NUMBER,job-name=compass-ord")
+fi
+
+POSITIONAL=()
+while [[ $# -gt 0 ]]
+do
+
+    key="$1"
+
+    case ${key} in
+        --image)
+            IMAGE="$2"
+            testCustomImage "${IMAGE}"
+            shift
+            shift
+            ;;
+        --*)
+            echo "Unknown flag ${1}"
+            exit 1
+            ;;
+        *)    # unknown option
+            POSITIONAL+=("$1") # save it in an array for later
+            shift # past argument
+            ;;
+    esac
+done
+set -- "${POSITIONAL[@]}" # restore positional parameters
+
+
+if [[ -z "$IMAGE" ]]; then
+    log::info "Provisioning vm using the latest default custom image ..."
+    
+    IMAGE=$(gcloud compute images list --sort-by "~creationTimestamp" \
+         --filter "family:custom images AND labels.default:yes" --limit=1 | tail -n +2 | awk '{print $1}')
+    
+    if [[ -z "$IMAGE" ]]; then
+       log::error "There are no default custom images, the script will exit ..." && exit 1
+    fi   
+ fi
+
+ZONE_LIMIT=${ZONE_LIMIT:-5}
+EU_ZONES=$(gcloud compute zones list --filter="name~europe" --limit="${ZONE_LIMIT}" | tail -n +2 | awk '{print $1}')
+
+for ZONE in ${EU_ZONES}; do
+    log::info "Attempting to create a new instance named compass-ord-test-${RANDOM_ID} in zone ${ZONE} using image ${IMAGE}"
+    gcloud compute instances create "compass-ord-test-${RANDOM_ID}" \
+        --metadata enable-oslogin=TRUE \
+        --image "${IMAGE}" \
+        --machine-type n1-standard-4 \
+        --zone "${ZONE}" \
+        --boot-disk-size 200 "${LABELS[@]}" &&\
+    log::info "Created compass-ord-test-${RANDOM_ID} in zone ${ZONE}" && break
+    log::error "Could not create machine in zone ${ZONE}"
+done || exit 1
+
+trap cleanup exit INT
+
+chmod -R 0777 /home/prow/go/src/github.com/kyma-incubator/compass/.git
+mkdir -p /home/prow/go/src/github.com/kyma-incubator/compass/components/console/shared/build
+
+log::info "Copying Compass to the instance"
+#shellcheck disable=SC2088
+utils::compress_send_to_vm "${ZONE}" "compass-ord-test-${RANDOM_ID}" "/home/prow/go/src/github.com/kyma-incubator/compass" "~/compass"
+
```

Comment:
Get the version that is listed in values.yaml for ORD service and checkout the branch for the listed PR there.